packages/eslint-plugin/docs/rules/no-uppercase-wrapper-types.mdx

```diff
@@ -0,0 +1,70 @@
+---
+description: 'Disallow using confusing built-in primitive class wrappers.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-uppercase-wrapper-types** for documentation.
+
+The JavaScript language includes a set of primitives such as `boolean` and `number`s.
+JavaScript also contains a "wrapper" class object for each of those primitives, such as [`Boolean`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) and [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number).
+TypeScript represents the primitive in lower-case types and the wrapper classes with upper-case
+
+Using the primitives like `0` instead of class constructors like `Number(0)` is generally considered a JavaScript best practice.
+As a result, using the lower-case primitive type names like `number` instead of class names like `Number` is generally considered a TypeScript best practice.
+
+Similarly, TypeScript's `Function` type allows any function-like value.
```

Comment:
Eh, I get we want to Keep It Stupid Simple I feel like this is just isn't quite right. 

Two ideas

***

Option 1. Try to preserve "Similarly" in order to attempt to tie these things together... 

`Similarly, "Function" refers to objects created with the "Function" constructor, i.e. "new Function("eval'd code with implicit args")", rather than ordinary function definitions. This is unsafe because yada yada yada`

***

Option 2. Just pivot.

`The Function type is unsound and permits being called with any number of arguments, yada yada yada`